plugins/auth-backend/src/providers/bitbucket/provider.ts

```diff
@@ -14,183 +14,15 @@
-  authHandler: AuthHandler<BitbucketOAuthResult>;
-  resolverContext: AuthResolverContext;
-};
-
-/** @public */
-export type BitbucketOAuthResult = {
-  fullProfile: BitbucketPassportProfile;
-  params: {
-    id_token?: string;
-    scope: string;
-    expires_in: number;
-  };
-  accessToken: string;
-  refreshToken?: string;
-};
-
-/** @public */
-export type BitbucketPassportProfile = PassportProfile & {
-  id?: string;
-  displayName?: string;
-  username?: string;
-  avatarUrl?: string;
-  _json?: {
-    links?: {
-      avatar?: {
-        href?: string;
-      };
-    };
-  };
-};
-
-export class BitbucketAuthProvider implements OAuthHandlers {
-  private readonly _strategy: BitbucketStrategy;
-  private readonly signInResolver?: SignInResolver<OAuthResult>;
-  private readonly authHandler: AuthHandler<OAuthResult>;
-  private readonly resolverContext: AuthResolverContext;
-
-  constructor(options: Options) {
-    this.signInResolver = options.signInResolver;
-    this.authHandler = options.authHandler;
-    this.resolverContext = options.resolverContext;
-    this._strategy = new BitbucketStrategy(
-      {
-        clientID: options.clientId,
-        clientSecret: options.clientSecret,
-        callbackURL: options.callbackUrl,
-        passReqToCallback: false,
-      },
-      (
-        accessToken: any,
-        refreshToken: any,
-        params: any,
-        fullProfile: passport.Profile,
-        done: PassportDoneCallback<OAuthResult, PrivateInfo>,
-      ) => {
-        done(
-          undefined,
-          {
-            fullProfile,
-            params,
-            accessToken,
-            refreshToken,
-          },
-          {
-            refreshToken,
-          },
-        );
-      },
-    );
-  }
-
-  async start(req: OAuthStartRequest): Promise<OAuthStartResponse> {
-    return await executeRedirectStrategy(req, this._strategy, {
-      accessType: 'offline',
-      prompt: 'consent',
-      scope: req.scope,
-      state: encodeState(req.state),
-    });
-  }
-
-  async handler(req: express.Request) {
-    const { result, privateInfo } = await executeFrameHandlerStrategy<
-      OAuthResult,
-      PrivateInfo
-    >(req, this._strategy);
-
-    return {
-      response: await this.handleResult(result),
-      refreshToken: privateInfo.refreshToken,
-    };
-  }
-
-  async refresh(req: OAuthRefreshRequest) {
-    const { accessToken, refreshToken, params } =
-      await executeRefreshTokenStrategy(
-        this._strategy,
-        req.refreshToken,
-        req.scope,
-      );
-    const fullProfile = await executeFetchUserProfileStrategy(
-      this._strategy,
-      accessToken,
-    );
-    return {
-      response: await this.handleResult({
-        fullProfile,
-        params,
-        accessToken,
-      }),
-      refreshToken,
-    };
-  }
-
-  private async handleResult(result: BitbucketOAuthResult) {
-    result.fullProfile.avatarUrl =
-      result.fullProfile._json!.links!.avatar!.href;
```

Comment:
We'll want to keep this logic, let's add it to the `defaultProfileTransform` where I mentioned it above.